### PR TITLE
Allow origins conf changes

### PR DESCRIPTION
**Pull request by mourya-33**
### Feature or Bugfix

- Feature

### Detail
This PR is to remove wildcards from origins as reported in https://github.com/data-dot-all/dataall/issues/1000 by using a configuration parameter from cdk.json. This will allow the users to set a custom origin (default is *)

### Relates
- https://github.com/data-dot-all/dataall/issues/1000

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes N/A
fetching data from storage outside the application (e.g. a database, an S3 bucket)? N/A
  - Is the input sanitized? N/A
  - What precautions are you taking before deserializing the data you consume? N/A
  - Is injection prevented by parametrizing queries? N/A
  - Have you ensured no `eval` or similar functions are used? N/A
- Does this PR introduce any functionality or component that requires authorization? N/A
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms? N/A
  - Are you logging failed auth attempts? N/A
- Are you using or adding any cryptographic features? N/A
  - Do you use a standard proven implementations? N/A
  - Are the used keys controlled by the customer? Where are they stored? N/A
- Are you introducing any new policies/roles/users? N/A
  - Have you used the least-privilege principle? How? N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mourya-33** [comment]
Hi @noah-paige here is the initial PR for allow-origins config driven change

**Comment by noah-paige** [comment]
@mourya-33 can you run `ruff format .` on the repo - looks like the GH workflows is failing

File: deploy/stacks/param_store_stack.py [hunk 1]
```diff
@@ -92,6 +93,13 @@ def __init__(
             parameter_name=f'/dataall/{envname}/pivotRole/enablePivotRoleAutoCreate',
             string_value=str(enable_pivot_role_auto_create),
         )
+        
+        aws_ssm.StringParameter(
+            self,
+            f'dataallAllowedOrigins{envname}',
+            parameter_name=f'/dataall/{envname}/allowedOrigins',
+            string_value=str(allowed_origins),
+        )
```
Comment (noah-paige) [inline deploy/stacks/param_store_stack.py:102]: do we need this SSM Parameter if just using env variable in Lambdas?

File: deploy/stacks/lambda_api.py [hunk 2]
```diff
@@ -161,6 +162,7 @@ def __init__(
             'envname': envname,
             'LOG_LEVEL': 'INFO',
             'email_sender_id': email_notification_sender_email_id,
+            'ALLOWED_ORIGINS': allowed_origins,
```
Comment (noah-paige) [inline deploy/stacks/lambda_api.py:165]: don't think we will need to set `ALLOWED_ORIGINS` for awshandler lambda... this lambda only reads from a sqs queue and is not handling API requests

File: deploy/stacks/lambda_api.py [hunk 3]
```diff
@@ -225,6 +227,7 @@ def __init__(
                 'custom_auth_url': custom_auth.get('url'),
                 'custom_auth_client': custom_auth.get('client_id'),
                 'custom_auth_jwks_url': custom_auth.get('jwks_url'),
+                'ALLOWED_ORIGINS': allowed_origins,
```
Comment (noah-paige) [inline deploy/stacks/lambda_api.py:230]: don't think we will need to set ALLOWED_ORIGINS for custom authorizer lambda... this lambda is for custom auth i not using cognito IdP and does not handle API requests

File: backend/api_handler.py
Comment (noah-paige) [inline backend/api_handler.py:1]: can we also do similar and set `'Access-Control-Allow-Origin': ALLOWED_ORIGINS,` in `dataall/backend/dataall/base/utils/api_handler_utils.py` for `def send_unauthorized_response()`

File: deploy/stacks/param_store_stack.py (hunk 1 above)
Comment (mourya-33) [reply deploy/stacks/param_store_stack.py:102]: Added this incase we want to use for local deployment testing. If this is not required for local deployment then i will remove this Noah.

File: deploy/stacks/lambda_api.py (hunk 2 above)
Comment (mourya-33) [reply deploy/stacks/lambda_api.py:165]: removed this Noah.

File: deploy/stacks/lambda_api.py (hunk 3 above)
Comment (mourya-33) [reply deploy/stacks/lambda_api.py:230]: Removed this as well.

File: backend/api_handler.py
Comment (mourya-33) [reply backend/api_handler.py:1]: Updated this to include the parameter

File: deploy/stacks/param_store_stack.py (hunk 1 above)
Comment (noah-paige) [reply deploy/stacks/param_store_stack.py:102]: for local testing allowed origins should just default to `*` in my opinion - think we can remove this SSM param

File: backend/dataall/modules/s3_datasets/cdk/dataset_stack.py
Comment (noah-paige) [inline backend/dataall/modules/s3_datasets/cdk/dataset_stack.py:1]: looks like the changes from glue security config encryption are being duplicated here from some type of merge conflict resolution

This is causing both `checkov` and `integration-tests` GH Workflows to fail

There should be no code changes to this file for this PR

File: deploy/stacks/param_store_stack.py (hunk 1 above)
Comment (mourya-33) [reply deploy/stacks/param_store_stack.py]: Removed this Noah

File: backend/dataall/modules/s3_datasets/cdk/dataset_stack.py
Comment (mourya-33) [reply backend/dataall/modules/s3_datasets/cdk/dataset_stack.py:1]: Fixed the merge conflict

**Comment by mourya-33** [comment]
> @mourya-33 can you run `ruff format .` on the repo - looks like the GH workflows is failing

Fixed the formatting Noah

File: deploy/stacks/backend_stack.py [hunk 4]
```diff
@@ -96,6 +97,7 @@ def __init__(
             reauth_apis=reauth_config.get('reauth_apis', None) if reauth_config else None,
             prod_sizing=prod_sizing,
             tooling_account_id=tooling_account_id,
+            allowed_origins=allowed_origins,
```
Comment (noah-paige) [inline deploy/stacks/backend_stack.py:100]: don't need to pass this param anymore

File: deploy/stacks/param_store_stack.py [hunk 5]
```diff
@@ -25,6 +25,7 @@ def __init__(
         reauth_apis=None,
         prod_sizing=False,
         tooling_account_id='',
+        allowed_origins='*',
```
Comment (noah-paige) [inline deploy/stacks/param_store_stack.py:28]: to remove

Comment (mourya-33) [reply deploy/stacks/param_store_stack.py]: removed

File: deploy/stacks/backend_stack.py (hunk 4 above)
Comment (mourya-33) [reply deploy/stacks/backend_stack.py]: removed

**Review by noah-paige — APPROVED** [review]
looks good - approving thanks @mourya-33

**Pull request merged**